src/crawlee/basic_crawler/_basic_crawler.py

```diff
@@ -723,24 +723,26 @@ async def _commit_request_handler_result(
         origin = context.request.loaded_url or context.request.url
 
         for call in result.add_requests_calls:
-            requests = list[BaseRequestData]()
+            requests = list[Request]()
 
             for request in call['requests']:
                 if (limit := call.get('limit')) is not None and len(requests) >= limit:
                     break
 
-                # If the request is a string, convert it to BaseRequestData object.
+                # If the request is a string, convert it to Request object.
                 if isinstance(request, str):
                     if is_url_absolute(request):
-                        dst_request = BaseRequestData.from_url(request)
+                        dst_request = Request.from_url(request)
 
                     # If the request URL is relative, make it absolute using the origin URL.
                     else:
                         base_url = call['base_url'] if call.get('base_url') else origin
                         absolute_url = convert_to_absolute_url(base_url, request)
-                        dst_request = BaseRequestData.from_url(absolute_url)
-                else:
-                    dst_request = request
+                        dst_request = Request.from_url(absolute_url)
+
+                # If the request is a BaseRequestData, convert it to Request object.
+                elif isinstance(request, BaseRequestData):
```

Comment:
I don't mind if this is defensive, just... how does that happen?

Reply:
![image](https://github.com/user-attachments/assets/832aab15-0883-4f12-8336-2b761ccd35fa)

I don't know - just covering all the possible cases Pylance declares.